src/models/vulnerability/source.ts

```diff
@@ -17,19 +17,31 @@ SPDX-License-Identifier: Apache-2.0
 Copyright (c) OWASP Foundation. All Rights Reserved.
 */
 
+import type { Comparable } from '../../_helpers/sortable'
+
 /** @beta */
 export interface OptionalSourceProperties {
   name?: Source['name']
   url?: Source['url']
 }
 
 /** @beta */
-export class Source {
+export class Source implements Comparable<Source> {
   name?: string
   url?: URL | string
 
   constructor (op: OptionalSourceProperties = {}) {
     this.name = op.name
     this.url = op.url
   }
+
+  compare (other: Source): number {
+    function normalizeUrl (u: URL | string): string {
+      return (typeof u === 'string') ? u : u.toString()
```

Comment:
not needed.
both, `string` and `URL` have a method `.toString()`
so just use the method, regardless of the type. 

see https://github.com/CycloneDX/cyclonedx-javascript-library/blob/1bb0aed3af79be3bd9e001c08251fbd290bd28cf/src/models/externalReference.ts#L43